### PR TITLE
`Offline Entitlements`: added support for fetching `ProductEntitlementMappingResponse` in `OfflineEntitlementsAPI`

### PR DESCRIPTION
**Pull request by NachoSoto**
### TODO:
- [x] Update snapshots

### CODE REVIEW

**Review by aboedo — APPROVED** [review]

File: Sources/Networking/HTTPClient/HTTPRequest.swift [hunk 1]
```diff
@@ -140,7 +142,7 @@ extension HTTPRequest.Path {
                 .postSubscriberAttributes,
                 .postAdServicesToken:
             return true
-        case .health:
```
Comment (tonidero) [inline Sources/Networking/HTTPClient/HTTPRequest.swift:143]: I guess this is because if this is going to live in a different server, we might not support etags in that place for now?

File: Sources/Networking/Operations/GetProductsEntitlementsOperation.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    override func begin(completion: @escaping () -> Void) {
+        self.getResponse(completion: completion)
+    }
+
+}
+
+private extension GetProductsEntitlementsOperation {
+
+    func getResponse(completion: @escaping () -> Void) {
+        let request = HTTPRequest(method: .get, path: .getProductsEntitlements)
```
Comment (tonidero) [inline Sources/Networking/Operations/GetProductsEntitlementsOperation.swift:50]: I'm not sure but we might have a different base URL for this one @aboedo? Or would we just use DNS to point to whatever server we want to? If we need a different base URL, we will need to add support for that (in Android, that was added for diagnostics). In any case, that should be done in a different PR.

**Review by tonidero — APPROVED** [review]

File: Sources/Networking/HTTPClient/HTTPRequest.swift (hunk 1 above)
Comment (NachoSoto) [reply Sources/Networking/HTTPClient/HTTPRequest.swift:143]: Exactly.

File: Sources/Networking/Operations/GetProductsEntitlementsOperation.swift (hunk 2 above)
Comment (NachoSoto) [reply Sources/Networking/Operations/GetProductsEntitlementsOperation.swift]: Looks like the goal is to have it as the same domain, but it's not built yet.

File: Sources/Networking/HTTPClient/HTTPRequest.swift [hunk 3]
```diff
@@ -138,7 +140,8 @@ extension HTTPRequest.Path {
                 .postOfferForSigning,
                 .postReceiptData,
                 .postSubscriberAttributes,
-                .postAdServicesToken:
+                .postAdServicesToken,
+                .getProductEntitlementMapping:
```
Comment (NachoSoto) [inline Sources/Networking/HTTPClient/HTTPRequest.swift:144]: @tonidero I enabled ETags for this again after talking to the server people. Once that works there, the SDK will support it.

Comment (tonidero) [reply Sources/Networking/HTTPClient/HTTPRequest.swift:144]: Thanks for letting me know. Will probably revert my PR since there won't be any requests using that right now... We can add it back in the future.
